src/rpc/rpc-handler.ts

```diff
@@ -90,7 +90,7 @@ export class RPCHandler {
    * @param   {Function} callback Will be invoked with the returned result or if the rpc failed
    *                              receives to arguments: error or null and the result
    */
-  public make (name: string, data: any, callback: (error: string, data: any) => void): Promise<any> | undefined {
+  public make (name: string, data: any, callback?: (error: string, data: any) => void): Promise<any> | undefined {
     if (typeof name !== 'string' || name.length === 0) {
```

Comment:
if (this.services.connection.isConnected) {
  if (callback) {
    callback(EVENT.CLIENT_OFFLINE)
  } else {
    return promise that rejects
  }
}

Reply:
What about the provide and unprovide calls when offline ?